src/test/java/api/loans/CheckInByBarcodeTests.java

```diff
@@ -1038,6 +1041,39 @@ public void overdueFineCalculatedCorrectlyWhenHourlyFeeFinePolicyIsApplied() {
       homeLocation.getJson().getString("name"), ownerId, feeFineId, 7.0));
   }
 
+  @Test
+  public void itemCheckedInEventIsPublished() {
+    final IndividualResource james = usersFixture.james();
+    final UUID checkInServicePointId = servicePointsFixture.cd1().getId();
+    final IndividualResource homeLocation = locationsFixture.basedUponExampleLocation(
+      item -> item.withPrimaryServicePoint(checkInServicePointId));
+    final IndividualResource nod = itemsFixture.basedUponNod(item ->
+      item.withPermanentLocation(homeLocation.getId()));
+
+    DateTime checkOutDate = new DateTime(2020, 1, 18, 18, 0, 0, UTC);
+    DateTime checkInDate = new DateTime(2020, 1, 22, 15, 30, 0, UTC);
+
+    checkOutFixture.checkOutByBarcode(nod, james, checkOutDate);
+
+    CheckInByBarcodeResponse checkInResponse = checkInFixture.checkInByBarcode(
+      new CheckInByBarcodeRequestBuilder()
+        .forItem(nod)
+        .on(checkInDate)
+        .at(checkInServicePointId));
+
+    JsonObject checkedInLoan = checkInResponse.getLoan();
+
+    List<JsonObject> publishedEvents = Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(FakePubSub::getPublishedEvents, hasSize(2));
+
+    JsonObject event = publishedEvents.get(1);
+
+    assertThat(event, isCheckedInEvent());
+    assertThat(new JsonObject(event.getString("eventPayload")),
+      isValidCheckedInEventPayload(checkedInLoan));
```

Comment:
Could this detail be better hidden by defining a matcher for the event?

Reply:
Done